File: data/multilang_comparison/program_julia/program_julia/src/func10.jl
include("program_julia_head.jl")
function func10(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array4 = func11(params0, get_path(), loopsFactor);
      array4.refC -= 1
      if array4.refC == 0
          empty!(array4.data)
          array4 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array96 = func12(params0, get_path(), loopsFactor);
      array96.refC -= 1
      if array96.refC == 0
          empty!(array96.data)
          array96 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array133 = func13(params0, get_path(), loopsFactor);
   return array133;
end

File: data/multilang_comparison/program_cpp_vector/src/func10.cpp
#include "program_cpp_vector.hpp" 
Array* func10(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array4 = func11(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array4->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array4->refC--;
      if(array4->refC == 0) {
         array4->data.clear();
         delete array4;
         DEBUG_FREE(array4->id);
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array65 = func12(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array65->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array65->refC--;
      if(array65->refC == 0) {
         array65->data.clear();
         delete array65;
         DEBUG_FREE(array65->id);
      }
   }
   Array_param params0;
   params0.size = 0;
   std::vector<Array*> data_params0(0);
   params0.data = data_params0;
   Array* array101 = func13(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array101->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   return array101;
}

